src/org/pahappa/systems/registrationapp/views/UserView.java

```diff
@@ -92,20 +122,41 @@ private void registerUser() {
                 } else System.out.print("This username is already in use. New "); 
             }
 
-            System.out.println("Date Of Birth (dd/mm/yyyy):");
-            String dateOfBirthString = scanner.nextLine().trim();
-            String[] dateData = dateOfBirthString.split("/");
-            int year = Integer.parseInt(dateData[2]) - 1900;
-            int month = Integer.parseInt(dateData[1]) - 1;
-            int date = Integer.parseInt(dateData[0]);
-            Date dateOfBirth = new Date(year,month,date);
+            boolean validDate = false;
+            Date dateOfBirth = null;
+            while (!validDate) {
+                System.out.println("Date Of Birth (dd/mm/yyyy):");
+                String dateOfBirthString = scanner.nextLine().strip();
+                while (!dateOfBirthString.matches("\\d{1,2}/\\d{1,2}/\\d{4}")) {
+                    System.out.println("The date of birth must match the format dd/mm/yyyy. Please try again: ");
+                    dateOfBirthString = scanner.nextLine().strip();
+                }
+                String[] dateData = dateOfBirthString.split("/");
+                int year = Integer.parseInt(dateData[2]);
+                int month = Integer.parseInt(dateData[1]) - 1;
+                while (month < 0 || month > 11) {
+                    System.out.println("Invalid month. Please try again: ");
+                    month = Integer.parseInt(dateData[1]) - 1;
+                }
+                int date = Integer.parseInt(dateData[0]);
+                boolean leapYear = (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
+                while (date < 1 || (month==1 && date>(leapYear?29:28)) || date > 31) {
+                    System.out.println("Invalid date. Please try again: ");
+                    date = Integer.parseInt(dateData[0]);
+                }
+                year -= 1900; //because of the documentation's date format
+                dateOfBirth = new Date(year,month,date);
+                if (Calendar.getInstance().getTime().before(dateOfBirth)) {
+                    System.err.println("You cannot be born in the future. Please choose another date.");                    
+                } else if (dateOfBirth.before(new Date(0,1,1))) {
+                    System.err.println("You cannot be born before 1900. Please choose another date.");                    
+                } else {
+                    validDate = true;
+                }
+            }
```

Comment:
The manual date parsing and validation could be simplified using existing date parsing libraries. You can check them out and see, they will simplify your work